lib/semantics/resolve-names.cc

```diff
@@ -2363,23 +2346,55 @@ Symbol &DeclarationVisitor::DeclareObjectEntity(
   return symbol;
 }
 
-void DeclarationVisitor::Post(const parser::DeclarationTypeSpec::Type &x) {
-  if (const auto *symbol{ResolveDerivedType(DeclTypeSpec::TypeDerived)}) {
-    GetDeclTypeSpec()->derivedTypeSpec().set_scope(*symbol->scope());
+void DeclarationVisitor::Post(const parser::IntrinsicTypeSpec::Character &x) {
+  if (!charInfo_.length) {
+    charInfo_.length = ParamValue{SomeExpr{
+        evaluate::AsExpr(evaluate::Constant<evaluate::SubscriptInteger>{1})}};
```

Comment:
Could also be written as `evaluate::Expr<SubscriptInteger>{1}` if you think that would be more clear.

Reply:
Yes, I've changed it, and added a way to construct ParamValue directly from an int.